appsec-kit-backend/src/main/java/com/vaadin/appsec/backend/AppSecDTOProvider.java

```diff
@@ -104,70 +111,184 @@ List<Vulnerability> getVulnerabilities() {
-                    if (vulnDevAnalysis != null) {
-                        vulnerabilityDTO.setDeveloperStatus(
-                                vulnDevAnalysis.getStatus());
-                        vulnerabilityDTO.setDeveloperAnalysis(
-                                vulnDevAnalysis.getDeveloperAnalysis());
-                        vulnerabilityDTO.setDeveloperUpdated(
-                                vulnDevAnalysis.getUpdated());
-                    }
+    private boolean isSameName(String depName, String vulnDepName) {
+        return depName.equals(vulnDepName);
+    }
 
-                    Set<String> urls = new HashSet<>();
-                    vuln.getReferences()
-                            .forEach(ref -> urls.add(ref.getUrl().toString()));
-                    vulnerabilityDTO.setReferenceUrls(urls);
+    private boolean isVersionAffected(String depVersion, List<String> versions,
+            List<Range> ranges) {
+        return includedInVersions(depVersion, versions)
+                || includedInRanges(depVersion, ranges);
+    }
+
+    private boolean includedInVersions(String depVersion,
+            List<String> versions) {
+        return versions != null && versions.contains(depVersion);
+    }
 
-                    vulnerabilityDTOs.add(vulnerabilityDTO);
+    private boolean includedInRanges(String depVersion, List<Range> ranges) {
+        if (ranges != null) {
+            DefaultArtifactVersion depArtifactVersion = new DefaultArtifactVersion(
+                    depVersion);
+            for (Range range : ranges) {
+                if (beforeLimits(depArtifactVersion, range)) {
+                    boolean vulnerable = false;
+                    for (Event event : sortEvents(range.getEvents())) {
+                        Optional<DefaultArtifactVersion> introduced = getVersionFromEvent(
+                                event, INTRODUCED);
+                        Optional<DefaultArtifactVersion> fixed = getVersionFromEvent(
+                                event, FIXED);
+                        Optional<DefaultArtifactVersion> lastAffected = getVersionFromEvent(
+                                event, LAST_AFFECTED);
+
+                        if (introduced.isPresent() && depArtifactVersion
+                                .compareTo(introduced.get()) >= 0) {
+                            vulnerable = true;
+                        } else if (fixed.isPresent() && depArtifactVersion
+                                .compareTo(fixed.get()) >= 0) {
+                            vulnerable = false;
+                        } else if (lastAffected.isPresent()
+                                && depArtifactVersion
+                                        .compareTo(lastAffected.get()) > 0) {
+                            vulnerable = false;
+                        }
+                    }
+                    if (vulnerable) {
+                        return true;
+                    }
```

Comment:
If an `event` in the for-loop turns `vulnerable` to `true`, the next one might turn it back to `false`. Is this intended?

Reply:
Yes. The events are in sorted order: `introduced`, `fixed`, `last_affected`. After the `introduced` event if a `fixed` event exists and the dependency version is greater (or equal) than the `fixed` event version then that means the dependency is not affected anymore. The same applies to the `last_affected` just there is no equal check.

Reply:
Now I understand, thanks!